src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/PrimitiveArrayMutatorFactory.java

```diff
@@ -145,38 +149,66 @@ private void extractRange(AnnotatedType type) {
 
       switch (type.getType().getTypeName()) {
         case "int":
-          minRange = inRange.map(InRange::min).orElse((long) Integer.MIN_VALUE);
-          maxRange = inRange.map(InRange::max).orElse((long) Integer.MAX_VALUE);
+          {
+            LongRange r =
+                RangeSupport.resolveIntegralRange(
+                    type, (long) Integer.MIN_VALUE, (long) Integer.MAX_VALUE);
+            minRange = r.min;
+            maxRange = r.max;
+          }
           break;
         case "long":
-          minRange = inRange.map(InRange::min).orElse(Long.MIN_VALUE);
-          maxRange = inRange.map(InRange::max).orElse(Long.MAX_VALUE);
+          {
+            LongRange r = RangeSupport.resolveIntegralRange(type, Long.MIN_VALUE, Long.MAX_VALUE);
+            minRange = r.min;
+            maxRange = r.max;
+          }
           break;
         case "short":
-          minRange = inRange.map(InRange::min).orElse((long) Short.MIN_VALUE);
-          maxRange = inRange.map(InRange::max).orElse((long) Short.MAX_VALUE);
+          {
+            LongRange r =
+                RangeSupport.resolveIntegralRange(
+                    type, (long) Short.MIN_VALUE, (long) Short.MAX_VALUE);
+            minRange = r.min;
+            maxRange = r.max;
+          }
           break;
         case "char":
           minRange = inRange.map(InRange::min).orElse((long) Character.MIN_VALUE);
           maxRange = inRange.map(InRange::max).orElse((long) Character.MAX_VALUE);
           break;
         case "float":
-          minFloatRange = inRangeFloat.map(FloatInRange::min).orElse(Float.NEGATIVE_INFINITY);
-          maxFloatRange = inRangeFloat.map(FloatInRange::max).orElse(Float.POSITIVE_INFINITY);
-          allowNaN = inRangeFloat.map(FloatInRange::allowNaN).orElse(false);
+          {
+            FloatRange r =
+                RangeSupport.resolveFloatRange(
+                    type, Float.NEGATIVE_INFINITY, Float.POSITIVE_INFINITY, false);
+            minFloatRange = r.min;
+            maxFloatRange = r.max;
+            allowNaN = r.allowNaN;
+          }
           break;
         case "double":
-          minDoubleRange = inRangeDouble.map(DoubleInRange::min).orElse(Double.NEGATIVE_INFINITY);
-          maxDoubleRange = inRangeDouble.map(DoubleInRange::max).orElse(Double.POSITIVE_INFINITY);
-          allowNaN = inRangeDouble.map(DoubleInRange::allowNaN).orElse(false);
+          {
+            DoubleRange r =
+                RangeSupport.resolveDoubleRange(
+                    type, Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, false);
+            minDoubleRange = r.min;
+            maxDoubleRange = r.max;
+            allowNaN = r.allowNaN;
+          }
           break;
         case "boolean":
           minRange = inRange.map(InRange::min).orElse(0L);
```

Comment:
Existinc code bug: `InRange` is not applicable to booleans, so booleans should always set min and max to 0 and 1.